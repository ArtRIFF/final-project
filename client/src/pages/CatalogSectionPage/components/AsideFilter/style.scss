@import '../../../../global_scss/mixins.scss';
@import '../../../../global_scss/vars.scss';

.aside-filter {
  position: absolute;
  width: inherit;
  &__header {
    @include flex-kit(space-between);
    margin-bottom: 30px;
  }

  &__title {
    @include font-value(700, 36px, 45px);
  }

  &__button {
    @include font-value(400, 24px, 150%);
    font-family: $secondary-font, sans-serif;
    color: rgba(49, 28, 28, 0.5);
    border: none;
    background-color: transparent;
    cursor: pointer;
  }

  &__wrapper {
    @include flex-kit(space-between);
    flex-direction: column;
  }
}

.filter-parameter {
  border-bottom: 1px solid rgba(49, 28, 28, 0.5);
  width: 100%;
  padding-top: 10px;

  &__header {
    @include flex-kit(space-between);
    margin-bottom: 12px;
  }

  &__title {
    @include font-value(500, 20px, 25px);
  }

  &__button {
    width: 24px;
    height: 24px;
    cursor: pointer;
    background-color: transparent;
    border: none;
    background-image: url('../../icons/arrow-down.svg');
    background-position: right;
    transform: rotate(180deg);
    background-size: 28px 23px;
    background-repeat: no-repeat;
  }

  &__container {
    @include font-value(400, 16px, 140%);
    font-family: $secondary-font, sans-serif;
    padding-left: 14px;

    label {
      margin-bottom: 18px;
    }
  }

  &__checkbox {
    @include flex-kit(flex-start);
    cursor: pointer;
  }

  &__checkbox span {
    margin-left: 14px;
  }

  &__container {
    max-height: 0;
    overflow: hidden;
    transition: max-height 0.2s ease-out;
  }
  

  &:not(.filter-parameter__hide) .filter-parameter__button {
    transform: rotate(360deg);
    transform: translateX(4px);
  }

  .price-container {
    flex-direction: row;
    input {
      width: 90px;
      border: 2px solid rgba(49, 28, 28, 0.7);
      border-radius: 2px;
      @include font-value(400, 16px, 140%);
      font-family: $secondary-font, sans-serif;
      margin-bottom: 10px;
    }
  }
}

input[type="checkbox"] {
  cursor: pointer;
  appearance: none;
  background-color: #fff;
  margin: 0;
  width: 24px;
  height: 24px;
  border: 2px solid rgba(49, 28, 28, 0.7);
  border-radius: 2px;
}

input[type="checkbox"]:checked {
  background-color: #311C1C;
  position: relative;
}

input[type="checkbox"]:checked::before {
  content: '';
  width: 30px;
  height: 30px;
  position: relative;
  display: block;
  background-image: url('../../icons/vector.svg');
  transform: rotate(0deg);
  left: 0;
  top: 3px;
  background-repeat: no-repeat;
}