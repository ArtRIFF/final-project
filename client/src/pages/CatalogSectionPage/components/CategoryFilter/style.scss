@import '../../../../global_scss/vars.scss';
@import '../../../../global_scss/mixins.scss';

.category-filter {
  @include flex-kit(space-between);
  width: inherit;
  height: inherit;

  font-family: $main-font__regular;
  @include font-value(300, 20px, 25px);
  
  &__result {
    @media (max-width: 460px) {
      display: none;
  }
  }

  &--btn {
    display: none;
    @media (max-width:$desktop-size) {
      display: inline-block;
  }
  }

  select {
    background-color: transparent;
    border: none;
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
    outline: none;
    cursor: pointer;
    appearance: none;
    background-image: url('../../icons/arrow-down.svg') ;
    background-position: right;
    background-repeat: no-repeat;
    padding-right: 25px;
  }

}