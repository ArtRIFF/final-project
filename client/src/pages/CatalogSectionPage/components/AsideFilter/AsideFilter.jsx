import './style.scss';
import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const AsideFilter = ({ filterRequest }) => {
  const url = window.location.search;
  const params = new URLSearchParams(url);

  const categoriesParam = params.get('categories')?.split(",");
  const metalParam = params.get('metal')?.split(",");
  const collectionParam = params.get('collectionName')?.split(",");
  const metalColorParam = params.get('metalColor')?.split(",");
  const insertTypeParam = params.get('insertType')?.split(",");
  const statusTypeParam = params.get('statusProduct')?.split(",");
  const minPriceParam = params.get('minPrice');
  const maxPriceParam = params.get('maxPrice');

  const [productsType, setProductsType] = useState(categoriesParam || []);
  const [metalType, setMetalType] = useState(metalParam || []);
  const [collectionType, setCollectionType] = useState(collectionParam || []);
  const [metalColorType, setMetalColorType] = useState(metalColorParam || []);
  const [insertType, setInsertType] = useState(insertTypeParam || []);
  const [statusType, setStatusType] = useState(statusTypeParam || []);
  const [minPrice, setMinPrice] = useState(minPriceParam || 0);
  const [maxPrice, setMaxPrice] = useState(maxPriceParam || 0);

  useEffect(() => {
    updateFilter();
    const value = document.querySelectorAll('.filter-parameter__checkbox');
    value.forEach(item => {

      productsType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });

      metalType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });

      collectionType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });

      metalColorType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });

      insertType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });

      statusType.forEach((param) => {
        if (item.dataset.categoryName === param) {
          item.querySelector('input').checked = true;
        }
      });
    })

  }, [productsType, metalType, collectionType, metalColorType, insertType, minPrice, maxPrice,statusType]);

  const updateFilter = () => {
    let filterParamArray = [];

    const pushToFilterParamArray = (param) => {
      if (param) {
        filterParamArray.push(param);
      }
    }

    const productsTypeParam = productsType.length ? `categories=${productsType.join(',')}` : "";
    pushToFilterParamArray(productsTypeParam);
    const metalTypeParam = metalType.length ? `metal=${metalType.join(',')}` : "";
    pushToFilterParamArray(metalTypeParam);
    const collectionTypeParam = collectionType.length ? `collectionName=${collectionType.join(',')}` : "";
    pushToFilterParamArray(collectionTypeParam);
    const metalColorTypeParam = metalColorType.length ? `metalColor=${metalColorType.join(',')}` : "";
    pushToFilterParamArray(metalColorTypeParam);
    const statusTypeParam = statusType.length ? `statusProduct=${statusType.join(',')}` : "";
    pushToFilterParamArray(statusTypeParam);
    const insertTypeParam = insertType.length ? `insertType=${insertType.join(',')}` : "";
    pushToFilterParamArray(insertTypeParam);
  
    let rangePriceParam = '';
    if (maxPrice && minPrice) {
      rangePriceParam = `maxPrice=${maxPrice}&minPrice=${minPrice}`;
    }
    if (maxPrice && !minPrice) {
      rangePriceParam = `maxPrice=${maxPrice}&minPrice=0`;
    }
    if ((!maxPrice && minPrice) || (!maxPrice && !minPrice)) {
      rangePriceParam = "";
    }

    pushToFilterParamArray(rangePriceParam);

    const filterParam = filterParamArray.length ? `${filterParamArray.join('&')}` : "";
    filterRequest(filterParam);
  };

  const onTypeChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    const arrayValue = value.split(' ');
    if (arrayValue.length > 1) {
      arrayValue.forEach(() => {
        setProductsType((!productsType.includes(arrayValue[0]) && checked)
          ? [...productsType, ...arrayValue]
          : productsType.filter(n => !arrayValue.includes(n))
        );
      })
    } else {
      setProductsType((!productsType.includes(value) && checked)
        ? [...productsType, value]
        : productsType.filter(n => n !== value)
      );
    }
  };

  const onMetalChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    setMetalType((!metalType.includes(value) && checked)
      ? [...metalType, value]
      : metalType.filter(n => n !== value)
    );
  };

  const onCollectionChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    setCollectionType((!collectionType.includes(value) && checked)
      ? [...collectionType, value]
      : collectionType.filter(n => n !== value)
    );
  };

  const onMetalColorChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    setMetalColorType((!metalColorType.includes(value) && checked)
      ? [...metalColorType, value]
      : metalColorType.filter(n => n !== value)
    );
  };

  const onStatusChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    setStatusType((!statusType.includes(value) && checked)
      ? [...statusType, value]
      : statusType.filter(n => n !== value)
    );
  };


  const onInsertChange = (e) => {
    const checked = e.target.checked;
    const value = e.target.closest('.filter-parameter__checkbox').dataset.categoryName;
    setInsertType((!insertType.includes(value) && checked)
      ? [...insertType, value]
      : insertType.filter(n => n !== value)
    );
  };

  const onMinPriceChange = ({ target: { value } }) => {
    if (Number.isFinite(+value)) {
      setMinPrice(value);
    }
  };

  const onMaxPriceChange = ({ target: { value } }) => {
    if (Number.isFinite(+value)) {
      setMaxPrice(value);
    }
  };

  const accordionAnimate = (e) => {
    const panels = [...e.target.closest('.aside-filter__wrapper').children];
    const indexTargetElem = panels.indexOf(e.target.closest('.filter-parameter'));
    panels.forEach((panel, i) => {
      if (i !== indexTargetElem && !panel.classList.contains('filter-parameter__hide')) {
        let panelContainer = panel.closest('.filter-parameter').children[1];
        panel.classList.add('filter-parameter__hide');
        panelContainer.style.maxHeight = null;
      }
    })

    e.target.closest('.filter-parameter').classList.toggle('filter-parameter__hide');
    const container = e.target.closest('.filter-parameter').children[1];
    if (container.style.maxHeight) {
      container.style.maxHeight = null;
    } else {
      container.style.maxHeight = container.scrollHeight + "px";
    }

  }

  const resetAllParam = () => {
    const checkboxElements = document.querySelectorAll('.filter-parameter__checkbox input');
    const inputElements = document.querySelectorAll('.price-container input');
    checkboxElements.forEach(checkbox => {
      checkbox.checked = false;
    })

    inputElements.forEach(input => {
      input.value = '';
    })

    setCollectionType([]);
    setProductsType([]);
    setMetalType([]);
    setMetalColorType([]);
    setInsertType([]);
    setMaxPrice(0);
    setMinPrice(0);
    setStatusType([]);

    updateFilter([]);
  }

  return (
    <div className="aside-filter">
      <div className='aside-filter__header'>
        <h3 className="aside-filter__title">Filters</h3>
        <button onClick={resetAllParam} className="aside-filter__button">Reset all</button>
      </div>
      <div className='aside-filter__wrapper'>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Price</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container price-container">
            <input onChange={onMinPriceChange} data-index={0} value={minPrice} type="text" placeholder={minPrice} />
            <span>-</span>
            <input onChange={onMaxPriceChange} data-index={1} value={maxPrice} type="text" placeholder={maxPrice} />
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Collection</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onClick={onCollectionChange} data-category-name={'vintage'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Vintage</span>
            </label>
            <label onClick={onCollectionChange} data-category-name={'seabed'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Seabed</span>
            </label>
            <label onClick={onCollectionChange} data-category-name={'refinement'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Refinement</span>
            </label>
            <label onClick={onCollectionChange} data-category-name={'capsule'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Capsule</span>
            </label>
            <label onClick={onCollectionChange} data-category-name={'max spass'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Max Spass</span>
            </label>
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Insert</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onClick={onInsertChange} data-category-name={'Without inserting'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Without insertion</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'pearl'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Pearl</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'zirconia'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Zirconia</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'diamond'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Diamond</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'Pearl and diamonds'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Pearl and diamonds</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'semi-precious'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Semi-precious</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'nacre'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Nacre</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'amethyst'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Amethyst</span>
            </label>
            <label onClick={onInsertChange} data-category-name={'onixl'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Onixl</span>
            </label>
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Metal</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onClick={onMetalChange} data-category-name={'Gold'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Gold</span>
            </label>
            <label onClick={onMetalChange} data-category-name={'Silver'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Silver</span>
            </label>
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Metal color</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onClick={onMetalColorChange} data-category-name={"White"} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>White</span>
            </label>
            <label onClick={onMetalColorChange} data-category-name={"Yellow"} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Yellow</span>
            </label>
            <label onClick={onMetalColorChange} data-category-name={"Red"} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Red</span>
            </label>
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Product Type</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onChange={onTypeChange} data-category-name={'bracelet'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Bracelet</span>
            </label>
            <label onChange={onTypeChange} data-category-name={'ring wedding-ring engagement-ring'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Ring</span>
            </label>
            <label onChange={onTypeChange} data-category-name={'earring child-earrings'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Earring</span>
            </label>
            <label onChange={onTypeChange} data-category-name={'cross'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Cross</span>
            </label>
            <label onChange={onTypeChange} data-category-name={'pendant'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Pendant</span>
            </label>
            <label onChange={onTypeChange} data-category-name={'pearl'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Pearl</span>
            </label>
          </div>
        </div>
        <div className="filter-parameter filter-parameter__hide">
          <div className="filter-parameter__header">
            <h5 className="filter-parameter__title">Product Status</h5>
            <button onClick={accordionAnimate} className="filter-parameter__button"></button>
          </div>
          <div className="filter-parameter__container">
            <label onChange={onStatusChange} data-category-name={'BESTSELLER'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Bestseller</span>
            </label>
            <label onChange={onStatusChange} data-category-name={'OUTLET'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>Outlet</span>
            </label>
            <label onChange={onStatusChange} data-category-name={'NEW'} className='filter-parameter__checkbox'>
              <input type="checkbox" />
              <span>New Collection</span>
            </label>
          </div>
        </div>
      </div>
    </div>
  );
};

AsideFilter.propTypes = {
  alreadyFilteredArray: PropTypes.array,
  filterRequest: PropTypes.func
};

export default AsideFilter;