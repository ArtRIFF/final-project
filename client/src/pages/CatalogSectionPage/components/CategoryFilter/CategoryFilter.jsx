import './style.scss';
import ButtonAll from '../../../../components/Button/ButtonAll/ButtonAll';
import PropTypes from 'prop-types';

const CategoryFilter = ({ onClickFunc, setResult, filterRequest }) => {

  const sortByNewest = () => {
    filterRequest("sort=NEW");
  }

  const sortByBestseller = () => {
    filterRequest("sort=BESTSELLER");
  }

  const sortByPrice = () => {
    filterRequest("sort=currentPrice");
  }

  const onSelectorChange = (e) => {
    switch (e.target.value) {
      case "price":
        sortByPrice();
        break;
      case "newest":
        sortByNewest();
        break;
      case "bestseller":
        sortByBestseller();
        break;
    }
  };
  return (
    <div className="category-filter">
      <div onClick={onClickFunc} className='category-filter--btn'>
        <ButtonAll text='Filter' className='section__btn-subscribe' />
      </div>
      <div className='category-filter__result'>Results: <span>{setResult}</span></div>
      <select data-testid='select-sortBy' onChange={onSelectorChange} name="select-sortBy" defaultValue={'DEFAULT'} >
        <option value="DEFAULT" disabled>Sort by</option>
        <option value="price">Price</option>
        <option value="newest">Newest</option>
        <option value="bestseller">Bestseller</option>
      </select>
    </div>
  );
};

CategoryFilter.propTypes = {
  onClickFunc: PropTypes.func,
  setResult: PropTypes.number,
  filterRequest: PropTypes.func
};

CategoryFilter.defaultProps = {
  onClickFunc: null
};

export default CategoryFilter;