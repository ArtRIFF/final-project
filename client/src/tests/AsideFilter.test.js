import AsideFilter from '../pages/CatalogSectionPage/components/AsideFilter/AsideFilter';
import CategoryFilter from '../pages/CatalogSectionPage/components/CategoryFilter/CategoryFilter';
import '@testing-library/jest-dom';
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

const onClickFunc = jest.fn();
const filterRequest = jest.fn();
const asideFilterRequest = jest.fn();

const renderAsideFilterComponent = () => {
  return (render(<AsideFilter
    filterRequest={asideFilterRequest}
    allCollectionArray={[
      {
        statusProduct: 'NEW',
        currentPrice: 12
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 6
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 3
      }
    ]}
  />))
}

const renderCategoryFilterComponent = () => {
  return (render(<CategoryFilter
    onClickFunc={onClickFunc}
    setResult={10}
    filterRequest={filterRequest}
    allCollectionArray={[
      {
        statusProduct: 'NEW',
        currentPrice: 12
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 6
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 3
      }

    ]}
    hasAnyFilters={true}
  />))
}

it("renders with snapshot", () => {
  const categoryFilter = renderer.create(<AsideFilter
    filterRequest={filterRequest}
    allCollectionArray={[]}
  />).toJSON()
  expect(categoryFilter).toMatchSnapshot();
});

it("test onclick event", () => {
  renderCategoryFilterComponent();
  renderAsideFilterComponent();
  fireEvent.click(screen.getByText(/Reset all/i))
  expect(asideFilterRequest).toHaveBeenCalled();
});