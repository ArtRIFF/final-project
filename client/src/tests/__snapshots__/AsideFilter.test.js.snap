// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
<div
  className="aside-filter"
>
  <div
    className="aside-filter__header"
  >
    <h3
      className="aside-filter__title"
    >
      Filters
    </h3>
    <button
      className="aside-filter__button"
      onClick={[Function]}
    >
      Reset all
    </button>
  </div>
  <div
    className="aside-filter__wrapper"
  >
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Price
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container price-container"
      >
        <input
          data-index={0}
          onChange={[Function]}
          placeholder="51"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          data-index={1}
          onChange={[Function]}
          placeholder="214453"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Collection
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name="vintage"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Vintage
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="seabed"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Seabed
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="refinement"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Refinement
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="capsule"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Capsule
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="max spass"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Max Spass
          </span>
        </label>
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Insert
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name="Without inserting"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Without insertion
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="pearl"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Pearl
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="zirconia"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Zirconia
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="diamond"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Diamond
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="Pearl and diamonds"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Pearl and diamonds
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="semi-precious"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Semi-precious
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="nacre"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Nacre
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="amethyst"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Amethyst
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="onixl"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Onixl
          </span>
        </label>
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Insert number
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container price-container"
      >
        <input
          onChange={[Function]}
          placeholder="1"
          type="text"
          value={0}
        />
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Metal
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name="Gold"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Gold
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="Silver"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Silver
          </span>
        </label>
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Metal color
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name="White"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            White
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="Yellow"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Yellow
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="Red"
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Red
          </span>
        </label>
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Sample
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name={575}
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            575
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name={585}
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            585
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name={750}
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            750
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name={958}
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            958
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name={985}
          onClick={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            985
          </span>
        </label>
      </div>
    </div>
    <div
      className="filter-parameter filter-parameter__hide"
    >
      <div
        className="filter-parameter__header"
      >
        <h5
          className="filter-parameter__title"
        >
          Product Type
        </h5>
        <button
          className="filter-parameter__button"
          onClick={[Function]}
        />
      </div>
      <div
        className="filter-parameter__container"
      >
        <label
          className="filter-parameter__checkbox"
          data-category-name="bracelet"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Bracelet
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="ring wedding-ring engagement-ring"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Ring
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="earring child-earrings"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Earring
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="cross"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Cross
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="pendant"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Pendant
          </span>
        </label>
        <label
          className="filter-parameter__checkbox"
          data-category-name="pearl"
          onChange={[Function]}
        >
          <input
            type="checkbox"
          />
          <span>
            Pearl
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
